***На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6.***
---
- Найти вероятность того, что выстрел произведен: 

 - a) первым спортсменом 
 - б) вторым спортсменом 
 - в) третьим спортсменом.

Будем считать по теореме Байеса

- H1 = три стрелка не попали в мишень
- H2 = все попали в мишень
- H3 = первый попал, другие не попали
- H4 = второй попал, другие не попали
- H5 = третий попал, другие не попали
- А - В мишень попал только один

1) Найдем вероятности этих событий

In [1]:
p1 = 0.9
q1 = 0.1
p2 = 0.8
q2 = 0.2
p3 = 0.6
q3 = 0.4 

p_H1 = q1 * q2 * q3
p_H1_A = 0

p_H2 = p1 * p2 * p3 
p_H2_A = 0

p_H3 = p1 * q2 * q3 
p_H3_A = 1

p_H4 = q1 * p2 * q3 
p_H4_A = 1

p_H5 = q1 * q2 * p3
p_H5_A = 1


p_H1, p_H2, p_H3, p_H4, p_H5

(0.008000000000000002,
 0.43200000000000005,
 0.07200000000000001,
 0.03200000000000001,
 0.012000000000000002)

2) Подставим в формулу Байеса значения, удовлетворяющие нас (там где А = 1)

In [6]:
first = (p_H3 * 1) / ((p_H1 * 0) + (p_H2 * 0) + (p_H3 * 1) + (p_H4 * 1) + (p_H5 * 1))
second = (p_H4 * 1) / ((p_H1 * 0) + (p_H2 * 0) + (p_H3 * 1) + (p_H4 * 1) + (p_H5 * 1))
third = (p_H5 * 1) / ((p_H1 * 0) + (p_H2 * 0) + (p_H3 * 1) + (p_H4 * 1) + (p_H5 * 1))

first, second, third

(0.6206896551724138, 0.2758620689655173, 0.10344827586206898)

Ответ: Вероятность что это был:
- Первый стрелок = 62%
- Второй стрелок = 27.58%
- Третий стрелок = 10.3%